crates/webidl/src/util.rs

```diff
@@ -693,6 +638,149 @@ impl<'src> FirstPassRecord<'src> {
+                            actual_signatures[j].args.push(idl_type.clone());
+                            first = false;
+                        } else {
+                            let mut sig = actual_signatures[j].clone();
+                            sig.args.truncate(i);
+                            sig.args.push(idl_type.clone());
+                            actual_signatures.push(sig);
+                        }
+                    }
+                }
+            }
+        }
+
+        let name = match id {
+            OperationId::Constructor => "new",
+            OperationId::Operation(Some(s)) => s,
+            OperationId::Operation(None) => {
+                warn!("unsupported unnamed operation");
+                return Vec::new()
+            }
+            OperationId::IndexingGetter => "get",
+            OperationId::IndexingSetter => "set",
+            OperationId::IndexingDeleter => "delete",
+        };
+
+        let mut ret = Vec::new();
+        for signature in actual_signatures.iter() {
+            // Ignore signatures with invalid return types
+            let ret_ty = match signature.orig.ret.to_idl_type(self) {
+                Some(ty) => ty,
+                None => continue,
+            };
+
+            let mut rust_name = name.to_snake_case();
+            let mut first = true;
+            for (i, arg) in signature.args.iter().enumerate() {
+                // Find out if any other known signature either has the same
+                // name for this argument or a different type for this argument.
+                let mut any_same_name = false;
+                let mut any_different_type = false;
+                let arg_name = signature.orig.args[i].name;
+                for other in actual_signatures.iter() {
+                    if other.orig.args.get(i).map(|s| s.name) == Some(arg_name) {
+                        if !ptr::eq(signature, other) {
+                            any_same_name = true;
+                        }
+                    }
+                    if other.args.get(i) != Some(arg) {
+                        any_different_type = true;
+                    }
+                }
+
+                // If all signatures have the exact same type for this argument,
+                // then there's nothing to disambiguate so we don't modify the
+                // name.
+                if !any_different_type {
+                    continue
+                }
+                if first {
+                    rust_name.push_str("_with_");
+                    first = false;
+                } else {
+                    rust_name.push_str("_and_");
+                }
+
+                // If this name of the argument for this signature is unique
+                // then that's a bit more human readable so we include it in the
+                // method name. Otherwise the type name should disambiguate
+                // correctly.
+                if !any_same_name {
```

Comment:
It is generally more readable to do `if foo { a() } else { b() }` rather than `if !foo { b() } else { a() }`

Reply:
Makes sense yeah (english made sense the other way to me) but in any case this ended up changing by the end